### ISSUE TITLE
Add a codespace file?

### ISSUE THREAD

**Issue by k-doering-NOAA** [NOAA-FIMS/case-studies#20]
If anyone would like to use it, should we add a devcontainer.json file for running a codespace into the case studies repo? I think dependencies needed would be:

- Quarto
- option to open R studio
- FIMS (built from main or from r universe)
- Other R packages listed

Maybe there are some other dependencies we are missing?

**Comment by iantaylor-NOAA** [comment]
What's the benefit of having a codespace file specific to case-studies, vs using the same codespaces setup for the main FIMS repo? I just worry about maintaining parallel things in two places.

**Comment by k-doering-NOAA** [comment]
I don't think we need all the tooling in the FIMS repo one - a lot of it is specific to building FIMS, which I don't think is needed if FIMS is being downloaded as a dependency. In addition, Quarto is needed here, but it isn't for building FIMS.

**Comment by ChristineStawitz-NOAA** [comment]
I think @MOshima-PIFSC tried to run her opaka paka case study within a codespace and ran into this, so seems like there is some interest in using codespaces within case-studies

**Comment by k-doering-NOAA** [comment]
I *think* I got codespaces working on this branch: https://github.com/NOAA-FIMS/case-studies/tree/fix-codespaces

The only thing missing is I couldn't install kaskr/TMB_contrib_R/TMBhelper from github, I think it caused the codespaces to crash. Perhaps it just needs to be installed in a different order? Not sure yet.

Note to use quarto, there is a vscode extension ("quarto/quarto"). Alternatively, start an rstudio session from the codespace by typing `rserver` into the bash, which will open up rstudio on a separate port (another tab of the browser)

@MOshima-PIFSC, when you have time, do you think you can try and see if 1) the codespace builds correctly and 2) if any features are missing?

**Comment by MOshima-PIFSC** [comment]
Thanks @k-doering-NOAA for working on this. I tried out the codespaces and was able to run my PFISC-opakapaka.r script successfully. However, I was not able to render the qmd files because there are some issues with mine and others that call `FIMSFrameAge()` which needs to be updated (see issue #25) and several of the scripts call {TMBhelper}. I think this package is called in the `PFISC-opakapaka.qmd`, `NWFSC-petrale.qmd`, and `AFSC-GOA-pollock.qmd` but only actually used in `AFSC-GOA-pollock.qmd`. I tried installing {TMBhelper} from the Rstudio session but it was unable to install. I also tried to install {tmbstan} because that is dependency and it failed to install as well (using `devtools::install_github("kaskr/tmbstan/tmbstan")` and the install packages GUI in RStudio). To clean up the case studies, we can remove the TMBhelper from opaka and petrale (maybe when @iantaylor-NOAA addresses issue #25) however, for Pollock, we may need to figure out how to install the package.  
I also tried to run gdbsource but was unable to, it doesn't look like it's included in the build setup. One reason I wanted to use a codespace for my casestudy was to help me debug because I could not get gdb properly set up on my local machine. If it's possible to add this in to the .devcontainer file, that would be really helpful! If it's a lot of work, maybe it can be added later on.

**Comment by iantaylor-NOAA** [comment]
Thanks for working on this @MOshima-PIFSC.
I've now removed FIMSFrameAge (#25 via PR #26) but did not touch the TMBhelper issue as I'm not sure how to deal with that.

**Comment by iantaylor-NOAA** [comment]
Clarification: I see that I could have just removed the unused `library(TMBhelper)` from petrale and opakapaka, but I think the best solution is figuring out how to get it installed so we can depend on it for the pollock case study within the codespace.

**Comment by k-doering-NOAA** [comment]
- I was able to get the codespace to install `TMBhelper`. For some reason, though, installing `reshape2` causes an error and I couldn't get it working. It's easily installed by using `install.packages("reshape2")` after the codespace opens, but this isn't ideal.
- I wasn't yet able to get gdb in the codespace file, but I was able to install it from the codespace using:
```
sudo apt-get update
sudo apt-get install gdb
```
I didn't test using it, though.

- I added in Ian's edits, so quarto compiling may work now?

**Comment by k-doering-NOAA** [comment]
@MOshima-PIFSC sorry this has lingered for so long!

Perhaps the best way forward for now is to merge in the codespaces files which is an improvement, and then open a new issue for the additional software that would be nice to add (getting the reshape2 package to install, adding gdb)

**Comment by k-doering-NOAA** [comment]
will be continued in #30
